### PR TITLE
Improve the error handling when parsing JSX

### PR DESCRIPTION
**Pull request by arunoda**
This improves the error handling when parsing JSX
* Throw an error if we found an unknown component
* Throw an error if we found a blocked script
* Use an Error object always when calling onError

### CODE REVIEW

File: source/components/JsxParser.js [hunk 1]
```diff
@@ -133,11 +133,20 @@ export default class JsxParser extends Component {
       .map(tag => tag.trim().toLowerCase()).filter(Boolean)
 
     if (/^(html|head|body)$/i.test(name)) return childNodes.map(c => this.parseElement(c))
-    if (blacklistedTags.indexOf(name.trim().toLowerCase()) !== -1) return undefined
+    const tagName = name.trim().toLowerCase()
+    if (blacklistedTags.indexOf(tagName) !== -1) {
+      onError(new Error(`The tag <${name}> is blocked, and will not be rendered.`))
```
Comment (TroyAlford) [inline source/components/JsxParser.js]: could you please update this to say `is blacklisted` instead of `is blocked`? :)

**Review by TroyAlford — CHANGES_REQUESTED** [review]
Man - this is some nice clean stuff! :) Thanks for the great PR!

File: source/components/JsxParser.js (hunk 1 above)
Comment (arunoda) [reply source/components/JsxParser.js]: @TroyAlford updated.

**Comment by TroyAlford** [comment]
Merged and published with release 1.17.0 :) Thanks for the submission!

**Pull request merged**